src/containers/App/index.js

```diff
@@ -3,18 +3,32 @@
  * App
  *
  */
-import React, { useReducer } from 'react';
+import React, { useReducer, useRef } from 'react';
 import { Switch, Route } from 'react-router-dom';
 import PropTypes from 'prop-types';
+import Strapi from '../../utils/Strapi';
 import Admin from '../Admin';
 import AppProvider from '../AppProvider';
 import AuthPage from '../AuthPage';
 import PrivateRoute from '../PrivateRoute';
+import StrapiProvider from '../StrapiProvider';
 import reducer, { initialState } from './reducer';
 import init from './init';
 
 function App({ appPlugins }) {
-  const [reducerState, dispatch] = useReducer(reducer, { initialState, appPlugins }, init);
+  // Initialize strapi api
+  const strapi = useRef(new Strapi());
```

Comment:
I wonder if ref here is necessary? Maybe you can just do const befor function declaration and use it inside a function? Probably it will be enough.

I just don't see any benefits of ref here, and official recomendation to refs is to not use them too often.

Reply:
I agree! it's an habit of using a ref to avoid passing a function to the dependencies array but in this case it's completely unnecessary I can declare it at the top of the file.